src/openrct2/drawing/LightFX.cpp

```diff
@@ -389,13 +389,14 @@ void lightfx_prepare_light_list()
 
         if (_current_view_zoom_front > 0)
         {
-            if ((entry->lightType & 0x3) < static_cast<int8_t>(_current_view_zoom_front))
+            if ((static_cast<uint8_t>(entry->lightType) & 0x3) < static_cast<int8_t>(_current_view_zoom_front))
             {
-                entry->lightType = LIGHTFX_LIGHT_TYPE_NONE;
+                entry->lightType = LightType::None;
                 continue;
             }
 
-            entry->lightType -= static_cast<int8_t>(_current_view_zoom_front);
+            entry->lightType = static_cast<LightType>(
```

Comment:
I'm not really a fan of this. Its because we are using the light type as a type and also a size. Could you create a function that can get the light type size and also a function that sets the light type size?
```
constexpr uint8_t GetLightTypeSize(LightType type){ return static_cast<uint8_t>(type) & 0x3; }
constexpr LightType SetLightTypeSize(LightType type, uint8_t size) { return static_cast<LightType>(((static_cast<uint8_t>(type) & ~0x3) | size); }
```

Reply:
Thank you! I was thinking it was a bit ugly that makes a lot of sense. When I add it to the commit should I credit you in the commit message or something like that?

Reply:
No need to.